File: test/SBRequest_iPad/SBRequest_iPad/FuckHttp.m
//
//  FuckHttp.m
//  testPOSTJSONinOC
//
//  Created by caoyuan on 7/8/14.
//  Copyright (c) 2014 caoyuan. All rights reserved.
//

#import "FuckHttp.h"
#import "FuckJSON.h"
@implementation FuckHttp

+(NSMutableDictionary*)post:(NSString*)url postDic:(NSMutableDictionary*)postData
{
    NSURL* URL = [NSURL URLWithString:url];
    NSMutableURLRequest* request = [[NSMutableURLRequest alloc]initWithURL:URL cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:30];
    [request setHTTPMethod:@"POST"];
    [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
    NSData * jsonDataFromDic = [FuckJSON dicToJSON:postData];
    [request  setHTTPBody:jsonDataFromDic];
    NSData *receiveNSData = [NSURLConnection sendSynchronousRequest:request returningResponse:nil error:nil];
    NSMutableDictionary *dic = [FuckJSON JSONtoDic:receiveNSData];
    return dic;

}

+(NSMutableDictionary*)get:(NSString*)url
{
    NSURL* URL = [NSURL URLWithString:url];
    NSMutableURLRequest* request = [[NSMutableURLRequest alloc]initWithURL:URL cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:30];
    [request setHTTPMethod:@"GET"];
    NSData *receiveNSData = [NSURLConnection sendSynchronousRequest:request returningResponse:nil error:nil];
    NSMutableDictionary *dic = [FuckJSON JSONtoDic:receiveNSData];
    return dic;

}

@end

File: test/ShanBay/ShanBay/FuckHttp.swift
//
//  FuckHttp.swift
//  test_Swfit_post_JSON
//
//  Created by caoyuan on 7/8/14.
//  Copyright (c) 2014 caoyuan_mini. All rights reserved.
//

import UIKit
//import "FuckHttp"

class FuckHttp: NSObject {

    class func post(url:NSString?, postData:NSDictionary?)->NSDictionary? {
        var URL = NSURL.URLWithString(url)
        var request = NSMutableURLRequest(URL: URL, cachePolicy: .UseProtocolCachePolicy, timeoutInterval: 10)
        request.HTTPMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        var jsonDataFromDic = FuckJSON.fuckEverythingToJson(postData)
        request.HTTPBody = jsonDataFromDic
        var receivedNSData = NSURLConnection.sendSynchronousRequest(request, returningResponse: nil, error: nil)
        var dic = FuckJSON.fuckJsonToDic(receivedNSData)
        return dic
    }

//    class func post(url:NSString?, postString:String?)->NSDictionary? {
//        var URL = NSURL.URLWithString(url)
//        var request = NSMutableURLRequest(URL: URL, cachePolicy: .UseProtocolCachePolicy, timeoutInterval: 10)
//        request.HTTPMethod = "POST"
//        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
//        var jsonDataFromDic = FuckJSON.fuckEverythingToJson(postString)
//        request.HTTPBody = jsonDataFromDic
//        var receivedNSData = NSURLConnection.sendSynchronousRequest(request, returningResponse: nil, error: nil)
//        var dic = FuckJSON.fuckJsonToDic(receivedNSData)
//        return dic
//    }

    class func get(url:NSString?)->NSDictionary? {
        var URL = NSURL.URLWithString(url)
        var receivedNSData = NSData.dataWithContentsOfURL(URL, options:NSDataReadingOptions.DataReadingUncached, error: nil)
        var dic = FuckJSON.fuckJsonToDic(receivedNSData)
        return dic

    }

}
